File: csharp/tests/JokeApi.Tests/ServiceTests.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using JokeApi.Repositories;
using JokeApi.Services;
using JokeApi.Models;
using System.IO;
using System.Threading.Tasks;

namespace JokeApi.Tests;

[TestClass]
public class ServiceTests
{
    private string CreateTempDbPath() => Path.Combine(Path.GetTempPath(), $"jokes_test_{Guid.NewGuid():N}.db");

    [TestMethod]
    public async Task ServiceCreatesAndFetches()
    {
        var db = CreateTempDbPath();
        var repo = new SqliteJokeRepository($"Data Source={db}");
        var svc = new JokeService(repo);
        var created = await svc.CreateAsync(new JokeCreate { Setup = "A", Punchline = "B" });
        var fetched = await svc.GetByIdAsync(created.Id);
        Assert.IsNotNull(fetched);
        Assert.AreEqual("A", fetched!.Setup);
    }
}

File: kotlin/app/src/test/kotlin/com/example/ServiceTests.kt
package com.example

import kotlin.test.*
import com.example.repositories.SqliteJokeRepository
import com.example.services.JokeService
import com.example.models.JokeCreate
import kotlinx.coroutines.runBlocking
import java.io.File

class ServiceTests {
    private fun tempDb(): String = File.createTempFile("jokes_test_", ".db").apply { delete() }.absolutePath

    @Test
    fun serviceCreateAndFetch() = runBlocking {
        val db = tempDb()
        val repo = SqliteJokeRepository("jdbc:sqlite:$db")
        val svc = JokeService(repo)
        val created = svc.create(JokeCreate(setup = "A", punchline = "B"))
        val fetched = svc.getById(created.id)
        assertNotNull(fetched)
        assertEquals("A", fetched!!.setup)
    }
}
